libbpf-rs/src/tc.rs

```diff
@@ -0,0 +1,150 @@
+use nix::errno;
+
+use crate::*;
+
+pub type TcAttachPoint = libbpf_sys::bpf_tc_attach_point;
+pub const BPF_TC_INGRESS: TcAttachPoint = libbpf_sys::BPF_TC_INGRESS;
+pub const BPF_TC_EGRESS: TcAttachPoint = libbpf_sys::BPF_TC_EGRESS;
+pub const BPF_TC_CUSTOM: TcAttachPoint = libbpf_sys::BPF_TC_CUSTOM;
+
+pub type TcFlags = libbpf_sys::bpf_tc_flags;
+pub const BPF_TC_F_REPLACE: TcFlags = libbpf_sys::BPF_TC_F_REPLACE;
+/// The BPF TC subsystem has different control paths from other BPF programs
+/// As such a BPF program using a TC Hook (SEC("classifier")) must be operated
+/// more independently from other [libbpf-rs::Program]s where the program itself can
+/// be attach_*() a program using TC must grab a hook point, or create one
+/// and attach from the TcHook
+///
+/// An example of using a BPF TC program can be seen in [examples/tc_whitelist_ports/src/main.rs]
+///
+/// Represents a Hook Point for a Traffic Control (TC) bpf program
+///
+/// This object exposes operations to create and attach a bpf_tc_hook
+/// into the TC subsystem
+#[derive(Clone, Copy)]
+pub struct TcHook {
+    hook: libbpf_sys::bpf_tc_hook,
+    opts: libbpf_sys::bpf_tc_opts,
+}
+
+impl TcHook {
+    /// Create a New TcHook given the file descriptor of the loaded SEC("classifier") Program
```

Comment:
s/classifier/tc/, it's a better and more modern naming

Reply:
+1

Reply:
The version of libbpf used by libbpf-sys does not have the necessary changes for using SEC("tc") yet.